test_cases/stdlib/test_unittest.py

```diff
@@ -0,0 +1,29 @@
+# pyright: reportUnnecessaryTypeIgnoreComment=true
+
+import unittest
+from datetime import datetime, timedelta
+from decimal import Decimal
+from fractions import Fraction
+
+case = unittest.TestCase()
+
+case.assertAlmostEqual(2.4, 2.41)
+case.assertAlmostEqual(Fraction(49, 50), Fraction(48, 50))
+case.assertAlmostEqual(datetime(1999, 1, 2), datetime(1999, 1, 2, microsecond=1), delta=timedelta(hours=1))
+case.assertAlmostEqual(datetime(1999, 1, 2), datetime(1999, 1, 2, microsecond=1), None, "foo", timedelta(hours=1))
+case.assertAlmostEqual(Decimal("1.1"), Decimal("1.11"))
+case.assertAlmostEqual(2.4, 2.41, places=8)
+case.assertAlmostEqual(2.4, 2.41, delta=0.02)
+case.assertAlmostEqual(2.4, 2.41, None, "foo", 0.02)
+
+case.assertAlmostEqual(2.4, 2.41, places=9, delta=0.02)  # type: ignore[call-overload]
+case.assertAlmostEqual("foo", "bar")  # type: ignore[call-overload]
+case.assertAlmostEqual(datetime(1999, 1, 2), datetime(1999, 1, 2, microsecond=1))  # type: ignore[arg-type]
+
+case.assertNotAlmostEqual(Fraction(49, 50), Fraction(48, 50))
+case.assertNotAlmostEqual(datetime(1999, 1, 2), datetime(1999, 1, 2, microsecond=1), delta=timedelta(hours=1))
+case.assertNotAlmostEqual(datetime(1999, 1, 2), datetime(1999, 1, 2, microsecond=1), None, "foo", timedelta(hours=1))
+
+case.assertNotAlmostEqual(2.4, 2.41, places=9, delta=0.02)  # type: ignore[call-overload]
+case.assertNotAlmostEqual("foo", "bar")  # type: ignore[call-overload]
+case.assertNotAlmostEqual(datetime(1999, 1, 2), datetime(1999, 1, 2, microsecond=1))  # type: ignore[arg-type]
```

Comment:
A couple thoughts about our tests in general:
- The tests work very well for pull requests like this one :) I was going to write my own tests as part of reviewing this, but it was nice that they already existed.
- I ran this test file with Python, making sure to get a runtime error whenever there's a type ignore comment. Ideally we would automate that somehow, but it doesn't seem to be worth the effort because we don't have that many tests.
- The error codes in tests are pretty much noise to me. I ignore them. I don't really care whether "wrong argument types" is a `call-overload` or `arg-type` error.

Reply:
> I ran this test file with Python, making sure to get a runtime error whenever there's a type ignore comment. Ideally we would automate that somehow, but it doesn't seem to be worth the effort because we don't have that many tests.

Oh, interesting idea. Tests like this file especially are basically very doctest-esque, just without the `>>>` prompt.

> The error codes in tests are pretty much noise to me. I ignore them. I don't really care whether "wrong argument types" is a call-overload or arg-type error.

Yes, and compounding this problem, there's the fact that we generally only include tests for the trickiest functions to write stubs for. These are usually the ones where mypy ends up emitting really weird error codes. E.g. loads of the ones in `test_sum.py` are `[list-item]`, and that's really meaningless; we shouldn't be asserting that mypy should emit such a meaningless error code (and there's no real reason why we should be giving mypy special treatment in these tests either). It seemed like a good idea at the time, but it's probably time to call it a day on the error-codes-in-tests.